File: ObjectAnimator/src/main/java/com/quyunshuo/objectanimator/MainActivity.java
package com.quyunshuo.objectanimator;

import android.animation.ObjectAnimator;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {
    private TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();

    }

    /**
     * 初始化View
     */
    private void initView() {
        textView = findViewById(R.id.text_view1);
        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showAnimator();
            }
        });
    }

    /***
     * 显示动画
     */
    private void showAnimator() {
////        5s内TextView变为全透明再变为原状
//        ObjectAnimator animator = ObjectAnimator.ofFloat(textView, "alpha", 1f, 0f, 1f);
//        animator.setDuration(5000);
//        animator.start();
//        让TextView旋转360度
        ObjectAnimator animator2 = ObjectAnimator.ofFloat(textView, "rotation", 0f, 360f);
        animator2.setDuration(1000);
        animator2.start();

    }
}

File: kotlinandroid/src/main/java/com/quyunshuo/kotlinandroid/MainActivity.kt
package com.quyunshuo.kotlinandroid

import android.animation.ObjectAnimator
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initView()
//        initAnimator()
//        test()
        whileTest()
        Log.d("MiYan", wenTest("3"))

    }

    private fun test() {

    }

    private fun initView() {
        text_view.setText(sum(10, 5))
        text_view.setOnClickListener { view ->
            //            initAnimator()
//            text_view.setText(splice("This ", "is ", "Kotlin"))
            text_view.setText(maxOf(3, 5).toString())

        }
    }

    private fun initAnimator() {
        var mObjectAnimator = ObjectAnimator.ofFloat(text_view, "rotation", 0f, 360f)
        mObjectAnimator.setDuration(1000)
        mObjectAnimator.start()
    }
//    fun sum(a: Int, b: Int): Int {
//        return a + b
//    }
//
//    fun main() {
//        print("sum of 3 and 5 is ")
//        println(sum(3, 5))
//    }

    private fun sum(a: Int, b: Int): String {
        var i = a + b
        return i.toString()
    }

    /**
     * 拼接字符串
     */
    private fun splice(s: String, t: String, r: String): String {
        val str: String = s + t + r
        return str
    }

    /**
     * 作为表达式的写法
     */
    private fun maxOf(a: Int, b: Int): Int = if (a > b) a else b

    private fun whileTest() {
        var items = listOf<String>("1", "2", "3")
        for (item in items) {
            Log.d("MiYan", item)
        }

        //返回索引
        for (index in items.indices) {
            Log.d("MiYan", "item at $index is ${items[index]}")
        }
        var a = 1
        var b = 2

        val c = if (a > b) {
            print("Choose a")
            a
        } else {
            print("Choose b")
            b
        }
        Log.d("MiYan", c.toString())
    }

    private fun wenTest(str: String): String {
        val string = when (str) {
            "1" -> "1"
            "2" -> "2"
            else -> "无"
        }
        return string
    }
}